cmd/thanos/receive.go

```diff
@@ -210,20 +206,43 @@ func runReceive(
 		})
 	}
 
-	level.Debug(logger).Log("msg", "setting up receive http handler")
+	level.Debug(logger).Log("msg", "setting up http handler")
 	{
-		ctx, cancel := context.WithCancel(context.Background())
-		g.Add(
-			func() error {
-				if err := webHandler.Run(ctx); err != nil {
-					return fmt.Errorf("error starting web server: %s", err)
-				}
-				return nil
-			},
-			func(err error) {
-				cancel()
-			},
-		)
+
+		mux := http.NewServeMux()
```

Comment:
Consider using router here instead of mux, for code consistency.

Reply:
I was just moving previous part of code around without bigger changes here.
Leaving like it is for now till we decide if we want to unify the mux/router thing.

Reply:
:+1:

Reply:
Ok, we can do this in a seperate PR